sumologic/resource_sumologic_collector.go

```diff
@@ -61,16 +61,15 @@ func resourceSumologicCollectorRead(d *schema.ResourceData, meta interface{}) er
 		collector, err = c.GetCollectorName(d.Id())
 		if err != nil {
 			log.Printf("[WARN] Collector not found when looking by name: %s, err: %v", d.Id(), err)
-			d.SetId("")
-			return err
+		} else if collector == nil {
+			log.Printf("[WARN] Got a nil Collector when looking by name: %s", d.Id())
```

Comment:
I think we still need `d.SetId("")` here? otherwise the provider will not clear state on refresh for resources that no longer exist

Reply:
I might not fully understand the domain here but what I'm trying to do here is to revert to version from before my last "fix" (https://github.com/SumoLogic/terraform-provider-sumologic/pull/92) - snippet below:

```go
        id, err := strconv.Atoi(d.Id())

        var collector *Collector
        if err != nil {
                collector, _ = c.GetCollectorName(d.Id())
                d.SetId(strconv.Itoa(collector.ID))
        } else {
                collector, _ = c.GetCollector(id)
        }

        if collector == nil {
                log.Printf("[WARN] Collector not found, removing from state: %v - %v", id, err)
                d.SetId("")

                return nil
        }
```

and:
* add some logging
* prevent from `panic`ing when `collector == nil` and we tried to call `d.SetId(strconv.Itoa(collector.ID))`, the `collector.ID` part is the one that would `panic`

The diff might not be 100% accurate here to reflect what was here before.

Reply:
Just to call this out explicitly: with the version from this PR we don't return in those `if`s hence we'll fall through to lines below which will do that:

```go
	if collector == nil {
		log.Printf("[WARN] Collector not found, removing from state: %v - %v", id, err)
		d.SetId("")
		return nil
	}
```

Reply:
ah ok - my bad, ya seems I was reading the diff wrong 🤔